src/main/java/school/lemon/changerequest/java/introduction/hw2/Task4.java

```diff
@@ -3,21 +3,22 @@
 public class Task4 {
     public static String charAt(String s, int i) {
 
-        if (s.equals(null)) return null;
+        if (s == null) return null;
 
-        if (i < 0 && -i <= (s.length() - 1)) {
-            int indexLastChar = s.length() + i;
-            char charAt = s.charAt(indexLastChar);
-            return Character.toString(charAt);
+        while (i < 0) {
+            i = i + s.length();
         }
-        if (i < 0 && -i > s.length()-1 || i > s.length()-1) {
-            while (true) {
 
-            }
-        } else return Character.toString(s.charAt(i));
+        while (i > s.length() - 1) {
+            i = i - s.length();
+        }
+
+        return Character.toString(s.charAt(i));
```

Comment:
Good implementation.
Just some parts could be shorter: Java support short style additions: `i+=s.length()` - it's same as `i=i+s.length()`.